### DISCUSSION TITLE
Using file paths

### DISCUSSION THREAD

**Discussion by timothy-barry** [nextflow-io/nextflow#1950] [Q&A] [answered]
Hello,

I am reading through the Nextflow documentation. I am confused by [something](https://www.nextflow.io/docs/latest/process.html#input-of-files) I read in Processes/Inputs/Input of files. In the example, `proteins` is defined in terms of a file path:
`proteins = Channel.fromPath( '/some/path/*.fa' )`.
But the tip at the bottom says, "This is also the reason why you should avoid whenever possible to use absolute or relative paths referencing files in your pipeline processes." So I have two questions.

1. When should I be using file paths? At the beginning of the script and never again?
2. What is the base directory of the file path  '/some/path/*.fa'? Is it stored in the `baseDir` global variable, as described [here](https://www.nextflow.io/docs/latest/script.html#script-implicit-variables)?

I am quite new to this, so thanks for the help.

**Comment by pditommaso** [comment, marked as answer]
The tip says "you should avoid whenever possible to use absolute or relative paths referencing files **in your pipeline processes**", which means in the task script, otherwise the task would be bound to a specific file (path).

> What is the base directory of the file path '/some/path/*.fa'? Is it stored in the `baseDir`

Doesn't make sense. The variable `baseDir` holds the path of the directory where the main pipeline script is located. It can be useful to reference any file that can be distributed along with the pipeline script. 

However, production data will be stored in a different path(s)
